### DISCUSSION TITLE
Generated null checks changed from 1.4.2 to 1.5.1 - How to get old behaviour back?

### DISCUSSION THREAD

**Discussion by otbe** [mapstruct/mapstruct#2877] [General]
Imagine we have a `Mapper` interface with a mapping method defined like this
```
@Mapper
interface GeneratedMapper {
  MyPojo map(
      OrganisationId organisation,
      UserId user,
      AuthenticationMethodsReferences amr,
      ZonedDateTime timestamp);
}

```

In the old 1.4.2 version of mapstruct this would generate the following code assuming we have a builder in `MyPojo`
```
@Override
public MyPojo map(AuthenticationMethodsReferences amr, UserId user, OrganisationId organisation, ZonedDateTime timestamp) {
    if ( amr == null && user == null && organisation == null && timestamp == null ) {
        return null;
    }

    MyPojo.Builder myPojo = MyPojo.builder();

    if ( amr != null ) {
        myPojo.amr( map( amr ) );
    }
    if ( user != null ) {
        myPojo.user( mapUserId( user ) );
    }
    if ( organisation != null ) {
        myPojo.organisation( map( organisation ) );
    }
    if ( timestamp != null ) {
        myPojo.timestamp( timestamp );
    }

    return myPojo.build();
}
```

However version 1.5.0 (and 1.5.1) generate something like this
```
@Override
public MyPojo map(OrganisationId organisation, UserId user, AuthenticationMethodsReferences amr, ZonedDateTime timestamp) {
    if ( organisation == null && user == null && amr == null && timestamp == null ) {
        return null;
    }

    MyPojo.Builder myPojo = MyPojo.builder();

    myPojo.organisation( map( organisation ) );
    myPojo.user( mapUserId( user ) );
    myPojo.amr( map( amr ) );
    myPojo.timestamp( timestamp );

    return myPojo.build();
}
```

The biggest problem for our use case is now that we have defaults in the builder class (these Pojos are generated by some external 3rd party tool). Without the additional null checks the methods like `timestamp` will always be called with null which leads to undesired side effects.

I scanned through the changelog and all new (and old) null related mapping settings but cant really relate my issue to something from the 1.5.0 release. 

So my question would be: is there a way to bring back the old null check behaviour or is there some reasoning behind this that I dont see right now?

Thank you very much :)

**Comment by filiphr** [comment]
Not sure what could have caused this. Usually we are never generating null checks for direct assignments. Only for implicit conversions, i.e. from `String` to `Integer`.

However, what you are looking for can be controlled through a `NullValueCheckStrategy`

e.g.

```java
@Mapper(nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS)
interface GeneratedMapper {
  MyPojo map(
      OrganisationId organisation,
      UserId user,
      AuthenticationMethodsReferences amr,
      ZonedDateTime timestamp);
}

```

**Reply by otbe** [reply]
Thank you very much.

Is there a way to set this for the whole project via compiler plugin setting or something like this?
It basically breaks a lot of projects and a lot of mappers everywhere for us 😅

**Comment by filiphr** [comment]
I had a better look at this one and I think that I do understand what you mean.

The null check you are referencing is for the source parameters. I'll create an issue so we can review this one.
